src/Account/TSPreKeyManager.m

```diff
@@ -241,11 +269,6 @@ + (void)refreshPreKeys {
                     DDLogDebug(@"%@ Not updating prekeys.", self.tag);
                 }
             }
-            
-            // Update the prekey check timestamp on success.
-            dispatch_async(TSPreKeyManager.prekeyQueue, ^{
-                lastPreKeyCheckTimestamp = [NSDate date];
-            });
```

Comment:
Previously we waited until the check (but not prekey) registration succeeded to mark the check as complete.  

* We were marking the check complete after stage 1 of the check but not after completion of the optional stage 2 of the check, so that wasn't great.
* Due to asynchronicity, this wouldn't de-bounce check requests that followed fast on each others' heels.
* We also want to re-check if any part of the registration fails.